### PR TITLE
Migrate to spdlog

### PR DESCRIPTION
**Pull request by l-laura**
This PR:
- adds a logging library based on the existing zswag implementation,
- changes log statements to spdlog,
- adds a test for file logging using the new environment variables,
- extends README with instructions for the developer setup and available flags/envvars.

Testing: existing tests should output over spdlog, and the file logger test should pass.

### CODE REVIEW

**Review by josephbirkner — COMMENTED** [review]
Nice, I like the decisions what is info and what is debug. Left a few more thoughts.

File: test/test-info.cpp [hunk 1]
```diff
@@ -2,14 +2,18 @@
...
+#include "mapget/log.h"
 
 using namespace mapget;
 
 TEST_CASE("InfoToJson", "[DataSourceInfo]")
 {
-    // Create a DataSourceInfo object
+    // Log all messages to the console if MAPGET_LOG_FILE is not specified.
+    if (getenv("MAPGET_LOG_FILE") == nullptr) {
+        setenv("MAPGET_LOG_LEVEL", "trace", 1);
```
Comment (josephbirkner) [inline test/test-info.cpp:13]: Maybe we can directly use `log()->set_level()` here?

**Review by josephbirkner — APPROVED** [review]
LGTM!

File: test/test-info.cpp (hunk 1 above)
Comment (l-laura) [reply test/test-info.cpp:18]: I will set the level to trace, since it's just test cases.

**Pull request merged**